dotnet/src/InternalUtilities/src/Diagnostics/ExceptionExtensions.cs

```diff
@@ -18,12 +18,10 @@ internal static class ExceptionExtensions
     /// <param name="ex">Exception.</param>
     /// <returns>True if <paramref name="ex"/> is a critical exception and should not be caught.</returns>
     internal static bool IsCriticalException(this Exception ex)
-        => ex is OutOfMemoryException
-            or ThreadAbortException
+        => ex is ThreadAbortException
```

Comment:
QQ, are we considering having this method in the Exception Handling overhaul?

Reply:
We discussed a that some of the exceptions can be removed from that list because they may never reach the code, but we have not considered removing it. @shawncal @stephentoub do you see any reason of keeping this method?

Reply:
Personally I'd suggest getting rid of it and not trying to special-case these.

Reply:
If there are no objections, I'll remove it in one of the next error-handling related PRs. Resolving for now.